test/integration/scheduler/workload_controller_test.go

```diff
@@ -69,6 +70,137 @@ var _ = ginkgo.Describe("Workload controller with scheduler", func() {
+		})
+
+		ginkgo.It("the workload should have appropriate AdditionalChecks added", func() {
+			wl := testing.MakeWorkload("wl", ns.Name).
+				Queue("queue").
+				Request(resourceGPU, "3").
+				Obj()
+			wlKey := client.ObjectKeyFromObject(wl)
+
+			ginkgo.By("creating and reserving quota the workload", func() {
+				gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+				gomega.Eventually(func() bool {
+					if err := k8sClient.Get(ctx, wlKey, &updatedWl); err != nil {
+						return false
+					}
+					return workload.HasQuotaReservation(&updatedWl)
+				}, util.Timeout, util.Interval).Should(gomega.BeTrue(), "should have quota reservation")
+			})
+
+			ginkgo.By("checking workload's admission checks", func() {
+				gomega.Eventually(func() []string {
+					gomega.Expect(k8sClient.Get(ctx, wlKey, &updatedWl)).To(gomega.Succeed())
+					return slices.Map(updatedWl.Status.AdmissionChecks, func(c *kueue.AdmissionCheckState) string { return c.Name })
+				}, util.Timeout, util.Interval).Should(gomega.ConsistOf("check1", "check2"))
+			})
+
+			ginkgo.By("updating admission check strategies", func() {
+				createdQueue := kueue.ClusterQueue{}
+				queueKey := client.ObjectKeyFromObject(clusterQueue)
+				gomega.Eventually(func() error {
+					gomega.Expect(k8sClient.Get(ctx, queueKey, &createdQueue)).To(gomega.Succeed())
+
+					createdQueue.Spec.AdmissionChecksStrategy.AdmissionChecks = []kueue.AdmissionCheckStrategyRule{
+						*testing.MakeAdmissionCheckStrategyRule("check1", flavorOnDemand).Obj(),
+						*testing.MakeAdmissionCheckStrategyRule("check2", reservationFlavor).Obj(),
+						*testing.MakeAdmissionCheckStrategyRule("check3").Obj()}
+					return k8sClient.Update(ctx, &createdQueue)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("checking workload's admission checks", func() {
+				gomega.Eventually(func() []string {
+					gomega.Expect(k8sClient.Get(ctx, wlKey, &updatedWl)).To(gomega.Succeed())
+					return slices.Map(updatedWl.Status.AdmissionChecks, func(c *kueue.AdmissionCheckState) string { return c.Name })
+				}, util.Timeout, util.Interval).Should(gomega.ConsistOf("check1", "check3"))
+			})
+
+			ginkgo.By("marking the checks as passed", func() {
+				gomega.Eventually(func() error {
+					gomega.Expect(k8sClient.Get(ctx, wlKey, &updatedWl)).To(gomega.Succeed())
+					workload.SetAdmissionCheckState(&updatedWl.Status.AdmissionChecks, kueue.AdmissionCheckState{
+						Name:    "check1",
+						State:   kueue.CheckStateReady,
+						Message: "check successfully passed",
+					})
+					workload.SetAdmissionCheckState(&updatedWl.Status.AdmissionChecks, kueue.AdmissionCheckState{
+						Name:    "check3",
+						State:   kueue.CheckStateReady,
+						Message: "check successfully passed",
+					})
+					return k8sClient.Status().Update(ctx, &updatedWl)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("admitting the workload", func() {
```

Comment:
You are not "admitting the workload" here, you are just checking.

Prefer to only use a new "by" when the test is changing something, not when it's checking an additional thing.

Reply:
Fixed

Reply:
Maybe you forgot to push?